lib/saluki-components/src/destinations/datadog_events_service_checks/mod.rs

```diff
@@ -0,0 +1,248 @@
+        let (requests_tx, requests_rx) = mpsc::channel(32);
+        // let events_service_checks = Metrics::from_component_context(context.component_context());
+        spawn_traced(run_io_loop(requests_rx, io_shutdown_tx, http_client));
+
+        debug!("Datadog Events and Service Checks destination started.");
+
+        while let Some(event_buffers) = context.events().next_ready().await {
+            debug!(event_buffers_len = event_buffers.len(), "Received event buffers.");
+
+            for event_buffer in event_buffers {
+                debug!(events_len = event_buffer.len(), "Processing event buffer.");
+
+                for event in event_buffer {
+                    match event.data_type() {
+                        DataType::EventD => {
+                            let request_builder = &mut events_request_builder;
+
+                            if let Some(eventd) = event.try_into_eventd() {
+                                let json = serde_json::to_string(&eventd).unwrap();
+
+                                match request_builder.create_request(json) {
+                                    Ok(request) => {
+                                        if requests_tx.send((1, request)).await.is_err() {
+                                            error!("Failed to send request to IO task: receiver dropped.");
+                                            return Err(());
+                                        }
+                                    }
+                                    Err(e) => {
+                                        error!(error = %e, "Failed to create request for event.");
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        DataType::ServiceCheck => {
+                            let request_builder = &mut service_checks_request_builder;
+                            if let Some(service_check) = event.try_into_service_check() {
+                                let json = serde_json::to_string(&service_check).unwrap();
+                                match request_builder.create_request(json) {
+                                    Ok(request) => {
+                                        if requests_tx.send((1, request)).await.is_err() {
+                                            error!("Failed to send request to IO task: receiver dropped.");
+                                            return Err(());
+                                        }
+                                    }
+                                    Err(e) => {
+                                        error!(error = %e, "Failed to create request for service check.");
+                                        continue;
+                                    }
+                                }
+                            }
+                        }
+                        _ => {}
```

Comment:
We should emit an error message here if we're getting events that aren't events or service checks.